libraries/AP_Soaring/SoaringController.cpp

```diff
@@ -0,0 +1,407 @@
+        _displayed_vario_reading = TE_FILT_DISPLAYED * _vario_reading + (1 - TE_FILT_DISPLAYED) * _displayed_vario_reading;
+        
+        Vector2f diff = location_diff(_prev_vario_update_location, current_loc); // get distances from previous update
+        float dx = diff.x;
+        float dy = diff.y;
+        Vector3f wind = _ahrs.wind_estimate();
+        float dx_w = wind.x * (AP_HAL::micros64() - _prev_vario_update_time) * 1e-6;
+        float dy_w = wind.y * (AP_HAL::micros64() - _prev_vario_update_time) * 1e-6;
+        dx -= dx_w;
+        dy -= dy_w;
+        _ekf_buffer[_ptr][0] = _vario_reading;
+        _ekf_buffer[_ptr][1] = dx;
+        _ekf_buffer[_ptr][2] = dy;
+        _ptr = (_ptr + 1) % EKF_MAX_BUFFER_SIZE;
+        _last_alt = _alt;                                       // Store variables
+        _last_roll=roll;
+        _last_aspd = aspd;
+        _last_total_E = total_E;
+        _prev_vario_update_location = current_loc;
+        _prev_vario_update_time = AP_HAL::micros64();
+        _new_data = true;
+        
+        log_vario_tuning.time_us = AP_HAL::micros64();
+        log_vario_tuning.aspd_raw = aspd;
+        log_vario_tuning.aspd_filt = _aspd_filt;
+        log_vario_tuning.alt = _alt;
+        log_vario_tuning.roll = roll;
+        log_vario_tuning.raw = _vario_reading;
+        log_vario_tuning.filt = _filtered_vario_reading;
+        log_vario_tuning.wind_x = wind.x;
+        log_vario_tuning.wind_y = wind.y;
+        log_vario_tuning.dx = dx;
+        log_vario_tuning.dy = dy;
+        log_vario_tuning.ptr = _ptr;
+        log_vario_tuning.head1 = HEAD_BYTE1;
+        log_vario_tuning.head2 = HEAD_BYTE2;
+        log_vario_tuning.msgid = _msgid2;
+        DataFlash_Class::instance()->WriteBlock(&log_vario_tuning, sizeof(log_vario_tuning));
+    }
+}
+
+float SoaringController::correct_netto_rate(float climb_rate, float phi, float aspd) {
+    // Remove aircraft sink rate
+    float CL0;  // CL0 = 2*W/(rho*S*V^2)
+    float C1;   // C1 = CD0/CL0
+    float C2;   // C2 = CDi0/CL0 = B*CL0
+    float netto_rate;
+    float cosphi;
+    CL0 = polar_K / (aspd * aspd);  
+    C1 = polar_CD0 / CL0;  // constant describing expected angle to overcome zero-lift drag
+    C2 = polar_B * CL0;    // constant describing expected angle to overcome lift induced drag at zero bank
+
+    cosphi = (1 - phi * phi / 2); // first two terms of mclaurin series for cos(phi)
+    netto_rate = climb_rate + aspd * (C1 + C2 / (cosphi * cosphi));  // effect of aircraft drag removed
+
+    // Remove acceleration effect - needs to be tested.
+    //float temp_netto = netto_rate;
+    //float dVdt = SpdHgt_Controller->get_VXdot();
+    //netto_rate = netto_rate + aspd*dVdt/GRAVITY_MSS;
+    //hal.console->printf_P(PSTR("%f %f %f %f\n"),temp_netto,dVdt,netto_rate,barometer.get_altitude());
+    return netto_rate;
+}
+ 
+float SoaringController::McCready(float alt) {
+    return thermal_vspeed; 
```

Comment:
Removed everything but that return statement, added a comment that this a method shell to be filled in later.